internal/container/reference.go

```diff
@@ -1,73 +1,112 @@
 package container
 
 import (
+	"fmt"
+
 	"github.com/docker/distribution/reference"
 	"github.com/pkg/errors"
 )
 
+// RefSet describes the references for a given image:
+//  1. ConfigurationRef: ref as specified in the Tiltfile
+//  2. LocalRef(): ref as used outside of the cluster (for Docker etc.)
+//  3. ClusterRef(): ref as used inside the cluster (in k8s YAML etc.). Often equivalent to
+//      LocalRef, but in some cases they diverge: e.g. when using a local registry with KIND,
+//      the image localhost:1234/my-image (localRef) is referenced in the YAML as
+//      http://registry/my-image (clusterRef).
 type RefSet struct {
 	// Ref as specified in Tiltfile; used to match a DockerBuild with
 	// corresponding k8s YAML. May contain tags, etc. (Also used as
 	// user-facing name for this image.)
 	ConfigurationRef RefSelector
 
-	// Image name as referenced from outside the cluster (in Dockerfile,
-	// docker push etc.). This will be the ConfigurationRef stripped of
-	// tags and (if applicable) prepended with the DefaultRegistry.
-	LocalRef reference.Named
-
-	// The image name (minus the Tilt tag) as referenced from the cluster (in k8s YAML,
-	// etc.) (Often this will be the same as the LocalRef, but in some cases they diverge:
-	// e.g. when using a local registry with KIND, the image localhost:1234/my-image
-	// (LocalRef) is referenced in the YAML as http://registry/my-image (clusterRef).
-	// clusterRef is optional; if not provided, we assume LocalRef == clusterRef.
-	clusterRef reference.Named
+	// (Optional) registry to prepend to ConfigurationRef to yield ref to use in update and deploy
+	Registry Registry
```

Comment:
i wonder if these should be private fields so that you can't change them after they've been validated